File: app/src/main/java/views/DislikeButton.java
package views;


import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.ImageView;

import com.sqube.tipshub.R;

@SuppressLint("AppCompatCustomView")
public class DislikeButton extends ImageView {
    String TAG = DislikeButton.class.getSimpleName();

    public static final int DISLIKED = 1;
    public static final int NOT_DISLIKED = -1;

    private int state;
    public DislikeButton(Context context) {
        super(context);
    }

    public DislikeButton(Context context, AttributeSet attr){
        super(context, attr);
    }

    private void init() {
        switch (state){
            case DISLIKED:
                setImageResource(R.drawable.ic_thumbs_down_color_alt);
                break;
            default:
                setImageResource(R.drawable.ic_thumbs_down_alt);
                break;
        }
    }


    public int getState() {
        return state;
    }

    public void setState(int state) {
        this.state = state;
        init();
    }
}

File: app/src/main/java/views/DislikeButton.kt
package views

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView
import views.DislikeButton
import com.sqube.tipshub.R

@SuppressLint("AppCompatCustomView")
class DislikeButton : ImageView {
    var TAG = DislikeButton::class.java.simpleName
    private var state = 0

    constructor(context: Context?) : super(context) {}
    constructor(context: Context?, attr: AttributeSet?) : super(context, attr) {}

    private fun init() {
        when (state) {
            DISLIKED -> setImageResource(R.drawable.ic_thumbs_down_color_alt)
            else -> setImageResource(R.drawable.ic_thumbs_down_alt)
        }
    }

    fun getState(): Int {
        return state
    }

    fun setState(state: Int) {
        this.state = state
        init()
    }

    companion object {
        const val DISLIKED = 1
        const val NOT_DISLIKED = -1
    }
}